frontend/src/metabase/visualizations/types/click-actions.ts

```diff
@@ -80,11 +81,17 @@ type UrlClickActionBase = {
 
 export type UrlClickAction = ClickActionBase & UrlClickActionBase;
 
+export type CustomClickAction = ClickActionBase & {
+  type: "custom";
+  onClick?: (closePopover: () => void) => void;
```

Comment:
The fact that the `onClick` handler accepts a `closePopover` method seems a bit strange to me, but should be fine for now.